integration/integration_test.go

```diff
@@ -88,185 +85,134 @@ func init() {
+		t.Cleanup(func() { s.log = nil })
+		test(t, s)
 	}
-	return t
-}
-
-// newTeleportIoT helper returns a running Teleport instance with Host as a
-// reversetunnel node.
-func (s *IntSuite) newTeleportIoT(c *check.C, logins []string) *TeleInstance {
-	// Create a Teleport instance with Auth/Proxy.
-	mainConfig := func() *service.Config {
-		tconf := s.defaultServiceConfig()
-		tconf.Auth.Enabled = true
-
-		tconf.Proxy.Enabled = true
-		tconf.Proxy.DisableWebService = false
-		tconf.Proxy.DisableWebInterface = true
-
-		tconf.SSH.Enabled = false
-
-		return tconf
-	}
-	main := s.newTeleportWithConfig(c, logins, nil, mainConfig())
-
-	// Create a Teleport instance with a Node.
-	nodeConfig := func() *service.Config {
-		tconf := s.defaultServiceConfig()
-		tconf.Hostname = Host
-		tconf.Token = "token"
-		tconf.AuthServers = []utils.NetAddr{
-			{
-				AddrNetwork: "tcp",
-				Addr:        net.JoinHostPort(Loopback, main.GetPortWeb()),
-			},
-		}
-
-		tconf.Auth.Enabled = false
-
-		tconf.Proxy.Enabled = false
-
-		tconf.SSH.Enabled = true
-
-		return tconf
-	}
-	_, err := main.StartReverseTunnelNode(nodeConfig())
-	c.Assert(err, check.IsNil)
-
-	return main
 }
 
 // newTeleportWithConfig is a helper function that will create a running
 // Teleport instance with the passed in user, instance secrets, and Teleport
 // configuration.
-func (s *IntSuite) newTeleportWithConfig(c *check.C, logins []string, instanceSecrets []*InstanceSecrets, teleportConfig *service.Config) *TeleInstance {
-	t := s.newTeleportInstance(c)
+func (s *integrationTestSuite) newTeleportWithConfig(t *testing.T, logins []string, instanceSecrets []*InstanceSecrets, teleportConfig *service.Config) *TeleInstance {
+	teleport := s.newTeleportInstance()
 
 	// use passed logins, but use suite's default login if nothing was passed
 	if len(logins) == 0 {
 		logins = []string{s.me.Username}
 	}
 	for _, login := range logins {
-		t.AddUser(login, []string{login})
+		teleport.AddUser(login, []string{login})
 	}
 
 	// create a new teleport instance with passed in configuration
-	if err := t.CreateEx(instanceSecrets, teleportConfig); err != nil {
-		c.Fatalf("Unexpected response from CreateEx: %v", trace.DebugReport(err))
+	if err := teleport.CreateEx(instanceSecrets, teleportConfig); err != nil {
+		t.Fatalf("Unexpected response from CreateEx: %v", trace.DebugReport(err))
 	}
-	if err := t.Start(); err != nil {
-		c.Fatalf("Unexpected response from Start: %v", trace.DebugReport(err))
+	if err := teleport.Start(); err != nil {
+		t.Fatalf("Unexpected response from Start: %v", trace.DebugReport(err))
 	}
 
-	return t
+	return teleport
 }
 
-// TestAuditOn creates a live session, records a bunch of data through it
+// TestNewStyleIntegrations acts as the master test suite for all integration tests
+// requiring standardised setup and teardown.
+func TestNewStyleIntegrations(t *testing.T) {
+	suite := newSuite(t)
+
+	t.Run("AuditOff", suite.bind(testAuditOff))
```

Comment:
I like getting away from check, but not sure about listing every test here. Is it possible to run one specific subtest with this?

Do we even need to have this group of tests in its own "suite"? If we just need a way to run them all together, would a common name prefix be enough so you could just do i.e. `go test ./integrations -run "TestIntegration.*"`?

Reply:
Perhaps, but I don't know enough about the structure to make those decisions yet. Basically, if a test was part of the `check` fixture, it became a subtest here.

Reply:
Also, being able to easily run a specific subtest was a strong goal for this change, so you can run a single test case via something like

```
$  go test -v ./integration -run TestIntegrations/Audit
# masses of log
--- PASS: TestIntegrations (4.69s)
    --- PASS: TestIntegrations/AuditOn (4.69s)
        --- PASS: TestIntegrations/AuditOn/recording_proxy,_sync_recording (4.69s)
PASS
ok      github.com/gravitational/teleport/integration   4.739s
```

Reply:
ahh I actually wasn't aware of that syntax for running subtests, that's sweet.

Yeah my open question then, for anyone who can answer, is if we really need to group all these tests into a single test/"suite" and explicitly list them all here or just break them into their own tests

Reply:
I am going to go with "for this PR, they need to stay together" as the change is already gigantic. But it's definitely something we should be looking at - I'm all for anything that makes these tests less scary.

Reply:
I think the reason @tcsc kept them together is to be able to execute common logic before each test akin to gocheck's SetUpTest? I think it's fine for now.

Reply:
Yeah. I wasn't sure which tests actually required that set up - some may not. I figured that the change was already huge enough without having to do any sort of detailed analysis and bring nuance into it as well. :-)

Hopefully this is a useful first step and we can refine it over time.

Reply:
yeah sounds good then